src/lib/components/Search/_styles.scss

```diff
@@ -0,0 +1,181 @@
+@import '../../../styles/settings/colors';
+@import '../../../styles/tools/mixins';
+
+// This represent the margin from the surrounding elements which needs to
+// be factored in when we're calculating the final width and translate
+// position for this element's expanded state.
+$WEB_SEARCH_MARGIN: 16px;
+
+// This accounts for the width of the close button which we display on mobile.
+// We need to offset the results popout by this amount so it doesn't appear
+// to extend past the search input field.
+$WEB_CLOSE_BUTTON_WIDTH: 45px;
+
+web-search {
+  --web-search-animation-time: 0;
+
+  align-items: center;
+  display: flex;
+  height: 100%;
+  justify-content: flex-end;
+  min-width: 50%;
+
+  @include bp(md) {
+    --web-search-animation-time: 200ms;
+    justify-content: flex-start;
+  }
+
+  .web-search__close-btn {
+    display: none;
+    margin-left: 8px;
+    text-overflow: initial;
+  }
+
+  .web-search__open-btn,
+  .web-search__close-btn {
+    border-radius: 0;
+    border: none;
+    box-shadow: none;
+    color: $GREY_700;
+    padding: 0 8px;
+    transition: none;
+
+    @include bp(md) {
+      display: none;
+    }
+  }
+
+  .web-search__open-btn:focus,
+  .web-search__open-btn:hover,
+  .web-search__close-btn:focus,
+  .web-search__close-btn:hover {
+    background: transparent;
+  }
+
+  .web-search__open-btn.focus-visible,
+  .web-search__close-btn.focus-visible {
+    // Use an inset box-shadow to draw the focus ring so we don't get weird
+    // clipping because we're adjacent to another element.
+    box-shadow: inset 0 0 0 1px $WEB_SECONDARY_COLOR;
+    outline: none;
+  }
+
+  .web-search__open-btn::before {
+    @include font-material-icon();
+    content: 'search';
+  }
+
+  .web-search__close-btn::before {
+    @include font-material-icon();
+    content: 'cancel';
+  }
+
+  .web-search__input-wrapper {
+    align-items: center;
+    color: $GREY_700;
+    display: none;
+    height: 100%;
+    width: 100%;
+
+    @include bp(md) {
+      display: flex;
+      max-width: calc(200% - #{$WEB_SEARCH_MARGIN} * 2);
+      min-width: calc(200% - #{$WEB_SEARCH_MARGIN} * 2);
+      // Display a minimum of 200px of search bar.
+      transform: translateX(calc(50% - 200px + #{$WEB_SEARCH_MARGIN}));
+      transition: var(--web-search-animation-time) transform;
```

Comment:
So this is fine and I'm not against the "read it in JS and match with a timeout" option but I wonder if this is better achieved with a "transitionend" listener. WDYT?

Reply:
yeah I wondered about using `transitionend` as well.

On desktop we do a transition, but we don't on mobile, so there's no `transitionend` event in that scenario. I just did [a quick check](https://jsbin.com/qalunin/2/edit?html,output) and it seems like if you do a 0s transition, Chrome doesn't fire a `transitionend` event.

If you know a workaround I'm happy to keep fiddling with it.